quinn-udp/src/unix.rs

```diff
@@ -795,6 +824,92 @@ fn decode_recv(
     })
 }
 
+#[cfg(target_os = "linux")]
+fn recv_err(io: SockRef<'_>) -> io::Result<Option<ICMPError>> {
+    use std::mem;
+
+    let fd = io.as_raw_fd();
+
+    let mut control = cmsg::Aligned([0u8; CMSG_LEN]);
+
+    // SEnding error info for now no data is recieved
+    let mut iovec = libc::iovec {
+        iov_base: std::ptr::null_mut(),
+        iov_len: 0,
+    };
+
+    let mut addr_storage: libc::sockaddr_storage = unsafe { mem::zeroed() };
+
+    // Have followed the previous declarations
+    let mut hdr: libc::msghdr = unsafe { mem::zeroed() };
+    hdr.msg_name = &mut addr_storage as *mut _ as *mut _;
+    hdr.msg_namelen = mem::size_of::<libc::sockaddr_storage>() as libc::socklen_t;
+    hdr.msg_iov = &mut iovec;
+    hdr.msg_iovlen = 1;
+    hdr.msg_control = control.0.as_mut_ptr() as *mut _;
+    hdr.msg_controllen = CMSG_LEN as _;
+
+    let ret = unsafe { libc::recvmsg(fd, &mut hdr, libc::MSG_ERRQUEUE) };
+
+    if ret < 0 {
+        let err = io::Error::last_os_error();
+        // EAGAIN/EWOULDBLOCK means no error in queue - this is normal
+        if err.kind() == io::ErrorKind::WouldBlock {
+            return Ok(None);
+        }
+        return Err(err);
+    }
+
+    let cmsg_iter = unsafe { cmsg::Iter::new(&hdr) };
+
+    for cmsg in cmsg_iter {
+        const IP_RECVERR: libc::c_int = 11;
+        const IPV6_RECVERR: libc::c_int = 25;
+
+        let is_ip_err = cmsg.cmsg_level == libc::IPPROTO_IP && cmsg.cmsg_type == IP_RECVERR;
+        let is_ipv6_err = cmsg.cmsg_level == libc::IPPROTO_IPV6 && cmsg.cmsg_type == IPV6_RECVERR;
+
+        if is_ip_err || is_ipv6_err {
```

Comment:
I'd suggest we flip this around: Negate it and use `continue`. That avoids right-wards drift of the code below.